relay-quotas/src/global.rs

```
@@ -0,0 +1,231 @@
+use std::sync::{Arc, Mutex, OnceLock, PoisonError};
+
+use relay_common::time::UnixTimestamp;
+use relay_redis::redis::Script;
+use relay_redis::{PooledClient, RedisError};
+
+use crate::RedisQuota;
+
+fn load_global_lua_script() -> &'static Script {
+    static SCRIPT: OnceLock<Script> = OnceLock::new();
+    SCRIPT.get_or_init(|| Script::new(include_str!("global_quota.lua")))
+}
+
+fn current_slot(window: u64) -> usize {
+    UnixTimestamp::now()
+        .as_secs()
+        .checked_div(window)
+        .unwrap_or_default() as usize
```

Should we log an error here? Might be weird to silently use a zero slot when the window is misconfigured to `0`.